README.md

```diff
@@ -57,17 +83,82 @@ In order to upload the file, initiating microservice or the client should send t
+
+When form will be submitted succesfully, the service will return HTTP 204 response with empty body. In case of any error,
+the service will return proper HTTP error response (4xx, 5xx) with details of the problem encoded as XML document. See Error handling
+section for details.
+
+When file is successfully uploaded by the user and checked by antivirus scanner, upscan service
+will make a POST call to the URL specified as a 'callbackUrl' parameter with the following body:
+
+```
+{
+    "reference" : "11370e18-6e24-453e-b45a-76d3e32ea33d",
+    "fileStatus" : "READY",
+    "url" : "https://bucketName.s3.eu-west-2.amazonaws.com?1235676"
+}
+```
+
+If the virus has been detected in the file or there is any other problem with the file. The following callback is being
+sent:
+
+```
+{
+    "reference" : "11370e18-6e24-453e-b45a-76d3e32ea33d",
+    "fileStatus" : "FAILED",
+    "details" : "The file has a virus"
+}
+```
+## Error handling
 
 In case of problems with uploading the file (file too small, too large, configuration problems), AWS
-will sent response that complies with this document: https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html
+will send response that complies with this document: https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html
+
+## Design considerations
+
+Service supports uploading single files only. If consuming service wants to upload multiple files during one user's journey,
+it has to make multiple independent calls to upscan-initiate.
+
+Callback URL provided by the service will be send in plain text through the network and visible to the end user. Because of that
+URL shouldn't point to the host that is publicly available from outside of MDTP network. It shouldn't also contain any 
+sensitive data (e.g. user identifiers, session tokens)
+The callback will be done from the inside of MDTP environment, so that when specifying
+callback URL.
+
+Upscan intentionally doesn't allow to attach any metadata nor tags to the uploaded file. It's expected that the service
+using upscan will keep track of requested uploads and keep the metadata internally.
+
+It's not recommended to use Upscan to route/transfer files between different services on the platform.
 
-# Additional notes
+# Architecture of the service
 
-The service is not implemented yet. Among others, storage of metadata, constraint validation, virus scanning and
-notifications using callbacks do not work yet.
+![Architecture](architecture.png)
 
-# Running locally
+The service heavily relies on Amazon Web Services to provide desired functionality. The most important
+AWS futures that are used by the service are:
+* [Pre-signed POST requests](https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-UsingHTTPPOST.html) - to allow end users to upload file directly to S3
+* [Pre-signed URLs](https://docs.aws.amazon.com/AmazonS3/latest/dev/ShareObjectPreSignedURL.html) - to allow consuming service to download uploaded file directly from S3
+
```

Comment:
Is it worth mentioning SQS queues here, or is that too much information for the end users?